modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -4,129 +4,186 @@ import cats.data.{EitherT, NonEmptyList, NonEmptySet}
 import cats.syntax.either._
+import cats.syntax.eq.catsSyntaxEq
 import cats.syntax.flatMap._
 import cats.syntax.functor._
 import cats.syntax.option._
 
+import org.tessellation.ext.crypto._
 import org.tessellation.kryo.KryoSerializer
 import org.tessellation.rosetta.domain._
-import org.tessellation.rosetta.domain.amount.{Amount, AmountValue}
+import org.tessellation.rosetta.domain.amount.Amount
+import org.tessellation.rosetta.domain.api.construction.ConstructionMetadata.MetadataResult
 import org.tessellation.rosetta.domain.api.construction.ConstructionParse
-import org.tessellation.rosetta.domain.currency.DAG
+import org.tessellation.rosetta.domain.api.construction.ConstructionPayloads.PayloadsResult
 import org.tessellation.rosetta.domain.error._
 import org.tessellation.rosetta.domain.operation._
-import org.tessellation.schema.transaction.{DAGTransaction, Transaction}
+import org.tessellation.schema.transaction._
 import org.tessellation.security.SecurityProvider
 import org.tessellation.security.hex.Hex
 import org.tessellation.security.key.ops._
 import org.tessellation.security.signature.Signed
 import org.tessellation.security.signature.signature.{Signature, SignatureProof}
 
 import eu.timepit.refined.auto._
-import eu.timepit.refined.boolean.Not
-import eu.timepit.refined.generic.Equal
-import eu.timepit.refined.refineV
 
 trait ConstructionService[F[_]] {
   def derive(publicKey: RosettaPublicKey): EitherT[F, ConstructionError, AccountIdentifier]
   def getAccountIdentifiers(operations: List[Operation]): Option[NonEmptyList[AccountIdentifier]]
   def getTransactionIdentifier(hex: Hex): EitherT[F, ConstructionError, TransactionIdentifier]
   def parseTransaction(hex: Hex, isSigned: Boolean): EitherT[F, ConstructionError, ConstructionParse.ParseResult]
   def combineTransaction(hex: Hex, signature: RosettaSignature): EitherT[F, ConstructionError, Hex]
+  def getPayloads(operations: NonEmptyList[Operation], metadata: MetadataResult): EitherT[F, ConstructionError, PayloadsResult]
 }
 
 object ConstructionService {
-  def make[F[_]: Async: SecurityProvider: KryoSerializer](): ConstructionService[F] = new ConstructionService[F] {
-    def derive(publicKey: RosettaPublicKey): EitherT[F, ConstructionError, AccountIdentifier] =
-      publicKey.hexBytes
-        .toPublicKeyByEC[F]
-        .map(_.toAddress)
-        .map(AccountIdentifier(_, None))
+  def make[F[_]: Async: SecurityProvider: KryoSerializer](saltGen: () => F[TransactionSalt]): ConstructionService[F] =
+    new ConstructionService[F] {
+      override def derive(publicKey: RosettaPublicKey): EitherT[F, ConstructionError, AccountIdentifier] =
+        publicKey.hexBytes
+          .toPublicKeyByEC[F]
+          .map(_.toAddress)
+          .map(AccountIdentifier(_, None))
+          .attemptT
+          .leftMap(_ => InvalidPublicKey)
+
+      override def getTransactionIdentifier(hex: Hex): EitherT[F, ConstructionError, TransactionIdentifier] = KryoSerializer[F]
```

Comment:
@marcinwadon It declares it is overriding the definition in the trait.

Reply:
It implements trait, not overrides it.

Reply:
IMO it makes the code safer and easier to read.  Before I added it I had to go look in the trait to see if this was something that was being overridden because everything was declared public. Further, the compiler will produce an error when nothing is being overridden which further helps verify we're writing the function we think we are.

Reply:
So IMO we should use the override modifier when it actually overrides something, not just for implementation (and this is what we do in the whole codebase actually)

Reply:
Are you referring to [SnapshotStorage.scala](https://github.com/Constellation-Labs/tessellation/blob/develop/modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/storage/SnapshotStorage.scala#L136)?   I don't see the issue.

I also don't understand your comment about failing with accidental overrides.